instrumentation/runtime-metrics/runtime-metrics-java17/library/src/main/java/io/opentelemetry/instrumentation/runtimemetricsjava17/RuntimeMetricsBuilder.java

```diff
@@ -28,36 +30,51 @@ public final class JfrTelemetryBuilder {
 
   /** Enable telemetry collection for all {@link JfrFeature}s. */
   @CanIgnoreReturnValue
-  public JfrTelemetryBuilder enableAllFeatures() {
+  public RuntimeMetricsBuilder enableAllFeatures() {
     Arrays.stream(JfrFeature.values()).forEach(this::enableFeature);
     return this;
   }
 
   /** Enable telemetry collection associated with the {@link JfrFeature}. */
   @CanIgnoreReturnValue
-  public JfrTelemetryBuilder enableFeature(JfrFeature feature) {
+  public RuntimeMetricsBuilder enableFeature(JfrFeature feature) {
     enabledFeatureMap.put(feature, true);
     return this;
   }
 
   /** Disable telemetry collection for all {@link JfrFeature}s. */
   @CanIgnoreReturnValue
-  public JfrTelemetryBuilder disableAllFeatures() {
+  public RuntimeMetricsBuilder disableAllFeatures() {
     Arrays.stream(JfrFeature.values()).forEach(this::disableFeature);
     return this;
   }
 
+  /** Disable telemetry collection for all metrics. */
+  @CanIgnoreReturnValue
+  public RuntimeMetricsBuilder disableAllMetrics() {
+    disableAllFeatures();
+    disableAllJmx();
+    return this;
+  }
+
   /** Disable telemetry collection associated with the {@link JfrFeature}. */
   @CanIgnoreReturnValue
-  public JfrTelemetryBuilder disableFeature(JfrFeature feature) {
+  public RuntimeMetricsBuilder disableFeature(JfrFeature feature) {
     enabledFeatureMap.put(feature, false);
     return this;
   }
 
-  /** Build and start an {@link JfrTelemetry} with the config from this builder. */
-  public JfrTelemetry build() {
+  /** Disable telemetry collection associated with the {@link JfrFeature}. */
+  @CanIgnoreReturnValue
+  public RuntimeMetricsBuilder disableAllJmx() {
```

Comment:
It would be great if we could integrate all the JMX into the `JfrFeature` enum (and rename it perhaps?); and whether we use JMX or JFR would become just an implementation detail.
That's an idea for a separate issue/PR though.

Reply:
Yes I think that's a good idea. It'll allow for more fine grained control over which JMX metrics get used as well.

Reply:
I'll make an issue for this one as well so we don't lose track of it.

Reply:
issue created here: https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/8473